modules/storage/minio.go

```diff
@@ -129,6 +129,7 @@ func NewMinioStorage(ctx context.Context, cfg interface{}) (ObjectStorage, error
 }
 
 func (m *MinioStorage) buildMinioPath(p string) string {
+	p = strings.TrimPrefix(p, "/") // url path prefix should not start with /
 	return util.PathJoinRelX(m.basePath, p)
```

Comment:
I think `util.PathJoinRelX` already removes the prefix slash. no need to trim it ahead?

Reply:
If `p = "/" `and `m.base = ""`, it generates "./" which is invalid to minio api, shows  `Received unexpected error:
  Object name cannot be empty`

Reply:
Nope, it never generates "./" here.

See the comment of "PathJoinRel" family functions,  it only outputs "", "." or "foo/bar" path

Even if you trim "/" here, if the input p is "./" , you still get "." and later your code append another suffix slash.

So ideally, I think it should be:

```
s = util.PathJoinRelX(m.basePath, p)
if p == "." {  p = "" }   // minio doesn't use dot as relative path
return p
```

I guess there could be a test case for it.

Reply:
`if p == "." {  p = "" } ` is ok